comp/remote-config/rcclient/rcclient.go

```diff
@@ -121,59 +114,38 @@ func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig,
 	}
 
 	// Checks who (the source) is responsible for the last logLevel change
-	// The priority between sources is: CLI > RC > Default
-	source, err := settings.GetRuntimeSource("log_level")
-	if err != nil {
-		pkglog.Errorf("Could not fetch source for 'log_level': %s", err)
-	}
+	source := config.Datadog.GetSource("log_level")
 
 	switch source {
-	case settings.SourceDefault, settings.SourceConfig:
-		// If the log level had been set by default
-		// and if we receive an empty value for log level in the config
-		// then there is nothing to do
-		if len(mergedConfig.LogLevel) == 0 {
-			return
-		}
-
-		// Get the current log level
-		var newFallback interface{}
-		newFallback, err = settings.GetRuntimeSetting("log_level")
-		if err != nil {
-			break
-		}
-
-		pkglog.Infof("Changing log level to '%s' through remote config", mergedConfig.LogLevel)
-		rc.configState.FallbackLogLevel = newFallback.(string)
-		// Need to update the log level even if the level stays the same because we need to update the source
-		// Might be possible to add a check in deeper functions to avoid unnecessary work
-		err = settings.SetRuntimeSetting("log_level", mergedConfig.LogLevel, settings.SourceRC)
-
-	case settings.SourceRC:
+	case model.SourceRC:
 		// 2 possible situations:
 		//     - we want to change (once again) the log level through RC
 		//     - we want to fall back to the log level we had saved as fallback (in that case mergedConfig.LogLevel == "")
-		var newLevel string
-		var newSource settings.Source
 		if len(mergedConfig.LogLevel) == 0 {
-			newLevel = rc.configState.FallbackLogLevel
-			// Regardless what the source was before RC override, we fallback to SourceConfig as it has now been changed by code
-			newSource = settings.SourceConfig
-			pkglog.Infof("Removing remote-config log level override, falling back to '%s'", newLevel)
+			pkglog.Infof("Removing remote-config log level override, falling back to '%s'", config.Datadog.Get("log_level"))
+			config.Datadog.UnsetForSource("log_level", model.SourceRC)
 		} else {
-			newLevel = mergedConfig.LogLevel
-			newSource = settings.SourceRC
+			newLevel := mergedConfig.LogLevel
+			newSource := model.SourceRC
 			pkglog.Infof("Changing log level to '%s' through remote config", newLevel)
+			err = settings.SetRuntimeSetting("log_level", newLevel, newSource)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 Nitpick:** ‏We can definitely simplify some variables here